test/deprecated/regexp_error.good

```diff
@@ -0,0 +1,15 @@
+regexp_error.chpl:5: warning: regexp: 'ok' is deprecated; exceptions are used to detect regexp compile errors
+regexp_error.chpl:12: warning: regexp: 'ok' is deprecated; exceptions are used to detect regexp compile errors
+regexp_error.chpl:14: warning: regexp: 'error()' is deprecated; exceptions are used to detect regexp compile errors
+regexp_error.chpl:23: warning: regexp: 'ok' is deprecated; exceptions are used to detect regexp compile errors
+false
+111111111111111111111
+222222222222222222222
+false
+333333333333333333333
+regexp: 'error()' is deprecated; exceptions are used to detect regexp compile errors
+444444444444444444444
+no argument for repetition operator: * when compiling regexp '*'
+555555555555555555555
+true
+re2/re2.cc:207: Error parsing '*': no argument for repetition operator: *
```

Comment:
This line from re2 seems out of place. Is there something we can adjust in how we call re2 to not print it?

Reply:
It seems to be printed by the RE2 library when a compilation error occurs. I don't know if we can suppress it.

Reply:
Also note, the `.good` file is based on a combination of `stdout` and `stderr` output, which can in principle be nondeterministically ordered.

Reply:
Lee, our tests need to have deterministic output when we match against the .good file, otherwise it will cause sporadic failures in nightly.  The program itself doesn't have to be deterministic, we have some tricks to massage the output so that it is ordered deterministically.  See [this part of the documentation](https://github.com/chapel-lang/chapel/blob/master/doc/rst/developer/bestPractices/TestSystem.rst#tests-with-varying-output)

Reply:
I have added `opts->set_log_errors(false);` to suppress RE2 error messages.

The test system seems deterministic enough; the `stderr` follows the `stdout`, as @ronawho describes.